laravel-app/tests/Unit/RegisterControllerTest.php

```diff
@@ -7,15 +7,58 @@
 use Tests\TestCase;
 use Illuminate\Foundation\Testing\Concerns\InteractsWithDatabase;
 use App\Http\Requests\CreateUserRequest;
+use App\Mail\EmailVerificationMail;
+use App\Models\User;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Mail;
+use Illuminate\Support\Str;
 
 class RegisterControllerTest extends TestCase
 {
     use RefreshDatabase;
     use WithFaker;
     use InteractsWithDatabase;
 
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        Mail::fake();
+    }
+
+    protected function assertEmailSent($email)
+    {
+        Mail::assertSent(EmailVerificationMail::class, function ($mail) use ($email) {
+            return $mail->hasTo($email);
+        });
+    }
+
+    protected function registerUser($userData)
+    {
+        $user = User::create([
+            'email' => $userData['email'],
+            'password' => Hash::make($userData['password']),
+            'username' => $userData['username'],
+            'email_verification_code' => Str::random(40),
+        ]);
+
+        $user->information()->create([
+            'first_name' => $userData['first_name'],
+            'last_name' => $userData['last_name'],
+        ]);
+
+        Mail::to($userData['email'])->send(new EmailVerificationMail($user));
+    }
```

Comment:
Why is your test not testing your service methods? non-sense right?? you are just testing the default creation of laravel.

Reply:
Apologies sir 🙇 will revise it now 🙇